### PR TITLE
3212 access-cloud-gov-environments-blocked-VPN

### LINKED ISSUE #3212: Bug - access to cloud.gov environments blocked by VPN
Thank you for taking the time to let us know about the issue you found. The basic rule for bug reporting is that 
something isn't working the way one would expect it to work. Please provide us with the information requested 
below and we will look at it as soon as we are able.

## Description

Raft has mandated VPN usage on company laptops, it appears that usage of the VPN occasionally blocks access to our development environments deployed in cloud.gov. A `403` error is returned and the following appears in the frontend logs

```bash
[error] 115#0: *15 access forbidden by rule, client: 2a09:bac0:1001:347::281:4d, server: , request: "GET / HTTP/1.1", host: "tdp-frontend-raft.app.cloud.gov"
```

## Action Taken

*In what way were you interacting with the application when you discovered the issue? Please be specific. Did it happen after you made a selection or clicked a button? Which page and which button? This information really helps us get to the bottom of an issue more quickly*

* Attempted to access `tdp-frontend-raft.app.cloud.gov`

## What I expected to see

*Please provide a short description of what you expected to see*

* The homepage

## What I did see

*Please provide a short description of what you did see. Screenshots are helpful, but please block out any personally identifying information before posting.*

* An HTTP `403` error returned by nginx

## Other Helpful Information

* The following IPv6 addresses are blocked

   ```
   2a09:bac0:1001:347::2dc:10
   2a09:bac0:1001:347::281:4d
   ```

* according to whatismyipaddress, the vpn is routing to New York, which should be encompassed in our nginx allow-list.
   image

* Turning off the VPN allows access to the site as expected
* It may be possible to exclude some domains from the VPN - do we need organizational support/permission?

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #3212 

## How to Test
_List the steps to test the PR_

1. Deploy to one of the apps
1. Browse the app
1. Verify that you can access deployed env
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by jtimpe — COMMENTED** [review]
failed on deploy: https://app.circleci.com/pipelines/github/raft-tech/TANF-app/29914/workflows/ccaa9104-9515-4413-8e5a-ca4ebee5e9f6/jobs/87946

```bash
   -----> Installing nginx 1.25.1
   Copy [/tmp/cache/final/dependencies/88daf4e79c4db8bf01ae3de3795661fd9a8b443ba4e498af6362be2d3e695a22/nginx_1.25.1_linux_x64_cflinuxfs4_c030b0cf.tgz]
BuildpackCompileFailed - App staging failed in the buildpack compile phase
FAILED
```

File: tdrs-frontend/nginx/cloud.gov/ip_whitelist_ipv6.conf [hunk 1]
```diff
@@ -537256,6 +537256,8 @@ allow 2A07:1500:C1C1:C1C1:C1C1:C100::/89;
 allow 2A07:1500:C1C1:C1C1:C1C1:C180::/90;
 allow 2A07:1500:C1C1:C1C1:C1C1:C1C0::/96;
 allow 2A07:1500:C1C1:C1C1:C1C1:C1C1::/97;
+allow 2a09:bac0:1001:347::2dc:10::/82
+allow 2a09:bac0:1001:347::281:4d::/82
```
Comment (jtimpe) [inline tdrs-frontend/nginx/cloud.gov/ip_whitelist_ipv6.conf:537260]: perhaps missing `;` at the end of the lines?

Comment (raftmsohani) [reply tdrs-frontend/nginx/cloud.gov/ip_whitelist_ipv6.conf:537260]: totally right

**Review by jtimpe — COMMENTED** [review]
still getting a deploy failure - not sure what caused this one

https://app.circleci.com/pipelines/github/raft-tech/TANF-app/29950/workflows/b753082c-5ed0-4ad2-86d8-ba8ae8502c90/jobs/88057?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary

```bash
Staging app and tracing logs...
   Cell 569c864c-7d32-4f47-8c4b-677b7119c890 creating container for instance e8a1a222-d940-4677-8c6a-73ed664988c6
   Security group rules were updated
   Cell 569c864c-7d32-4f47-8c4b-677b7119c890 successfully created container for instance e8a1a222-d940-4677-8c6a-73ed664988c6
   Downloading app package...
   Downloading build artifacts cache...
   Downloaded build artifacts cache (2.8M)
   Downloaded app package (9.3M)
   -----> Download go 1.19
   -----> Running go build supply
   /tmp/buildpackdownloads/2e4d3b83bf6f6915 ~
BuildpackCompileFailed - App staging failed in the buildpack compile phase
FAILED

Exited with code exit status 1
```

**Review by jtimpe — APPROVED** [review]

**Comment by raftmsohani** [comment]
tested and it worked for Jan!

**Comment by elipe17** [comment]
I am still blocked when my VPN is on. Looks like the cidr isn't inclusive?
Screenshot 2024-10-16 at 11 49 41 AM

**Comment by elipe17** [comment]
> I am still blocked when my VPN is on. Looks like the cidr isn't inclusive? Screenshot 2024-10-16 at 11 49 41 AM

Logs for SA. Just in case.

```
2024/10/16 15:55:21 [error] 115#0: *52 access forbidden by rule, client: 2a09:bac0:1001:347::2dc:8, server: , request: "GET / HTTP/1.1", host: "tdp-frontend-raft.app.cloud.gov"
11:55:21.969: [RTR.1] tdp-frontend-raft.app.cloud.gov - [2024-10-16T15:55:21.750205579Z] "GET / HTTP/1.1" 403 0 555 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36" "127.0.0.1:59448" "10.10.2.10:61060" x_forwarded_for:"2a09:bac0:1001:347::2dc:8, 127.0.0.1" x_forwarded_proto:"https" vcap_request_id:"9cbbb801-cfe0-4aaf-6ec6-66bd764ccae2" response_time:0.218592 gorouter_time:0.000078 app_id:"fbca8d0a-ea1e-4a32-ac13-b99df6af7808" app_index:"0" instance_id:"28d1948a-95a8-4407-4eca-b8c7" x_cf_routererror:"-" x_b3_traceid:"9cbbb801cfe04aaf6ec666bd764ccae2" x_b3_spanid:"6ec666bd764ccae2" x_b3_parentspanid:"-" b3:"9cbbb801cfe04aaf6ec666bd764ccae2-6ec666bd764ccae2"
```

**Comment by raftmsohani** [comment]
> I am still blocked when my VPN is on. Looks like the cidr isn't inclusive? Screenshot 2024-10-16 at 11 49 41 AM

@elipe17  Wait for Alex to test and redeploy to raft then test

**Review by elipe17 — APPROVED** [review]

**Pull request merged**